model/registry/http.yaml

```diff
@@ -95,6 +95,13 @@ groups:
           was the cause of the resending (e.g. redirection, authorization failure, 503 Server Unavailable, network issues,
           or any other).
         examples: 3
+      - id: request.size
```

Comment:
I think `http.request.envelop.size` would be more descriptive and would be better aligned with existing [`messaging.message.envelope.size`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/messaging.md)

Reply:
I think the idea was not to go far away from ECS naming, where we have already similar breaking change for `http.request|response.body.bytes` to `http.request|response.body.size`

Simple renaming from `bytes` to `size` is preferable in this case. Envelope would introduce a new entity

Reply:
Since we're introducing a breaking change anyway (sorry!), can we take this additional effort to make attribute name more descriptive?

If we decide to go with `http.request.size`  we'd either end up with less descriptive `messaging.message.size` and so on for the sake of consistency or will be inconsistent between conventions.

Reply:
Is there anything gained by having this consistency?  

In the HTTP space, there isn't anything that references an `envelope`, there are `message`s, `request`s, `response`s and http/2 introduces `frame`s.  So if we called this `envelope.size` someone who is familiar with HTTP would have to go and figure out what that actually means.

I'm not familiar with messaging systems, but I would guess that envelope has a special meaning within that ecosystem so that it is recognizable. 

If both of these ecosystems use different language to describe a similar thing, shouldn't we use the language that makes sense within that ecosystem?

Reply:
The consistency is not essential. But the reason to add `envelope` is to provide clear separation between the content size and the full size in the attribute name. 
`http.request.size` alone does not provide enough clarity. 

So I'm totally open to explore other more HTTP friendly names. Seems that [HTTP/2 friendly](https://web.dev/articles/performance-http2) one would be `message`, but I don't think `http.request.message.size` would emphasize that it's a full size.

Other alternatives I can think of  `http.request.full|total.size`.

Reply:
> The consistency is not essential. But the reason to add `envelope` is to provide clear separation between the content size and the full size in the attribute name.
> `http.request.size` alone does not provide enough clarity.

@lmolkova why do you think that `http.request.size` alone does not provide enough clarity?
While trying to be as precise as possible, I think we should also strive for simple names. Names cannot carry all the semantics anyways, that's why we have a descriptions that make the semantics explicit. And in this case `http.request.size` is quite self-explanatory as it is the `size of the request` (and since it's not further specified an HTTP request includes ALL of the request). While `http.request.body.size` is the `size of the request body`.

Since `envelope` is an explicit concept in messaging (that does not have a widely used corresponding term with HTTP) I think it's fine to have it explicit for messaging use cases (i.e. `messaging.message.envelope.size`).

Reply:
I think we're in the personal preference territory here and I don't want to block this based on my personal preferences. Let's go ahead with `http.request|response.size` especially given it's been out in ECS for a while.